Test `TestReplay/testdata/process_plugin_sqlc_gen_json` fails if `sqlc-gen-json` isn't found in `$PATH`

### Version

1.16.0

### What happened?

I cloned the repo and ran the tests, and `TestReplay/testdata/process_plugin_sqlc_gen_json` failed. I would expect the tests to pass

### Relevant log output

```shell
--- FAIL: TestReplay (0.12s)
    --- FAIL: TestReplay/testdata/process_plugin_sqlc_gen_json (0.01s)
        endtoend_test.go:118: sqlc generate failed: # package jsonb
            error generating code: process: sqlc-gen-json not found
FAIL
FAIL	github.com/kyleconroy/sqlc/internal/endtoend	2.834s
```


### Database schema

_No response_

### SQL queries

_No response_

### Configuration

_No response_

### Playground URL

_No response_

### What operating system are you using?

Linux

### What database engines are you using?

_No response_

### What type of code are you generating?

_No response_

--- a/internal/endtoend/endtoend_test.go
+++ b/internal/endtoend/endtoend_test.go
@@ -5,6 +5,7 @@ import (
 	"context"
 	"encoding/json"
 	"os"
+	osexec "os/exec"
 	"path/filepath"
 	"strings"
 	"testing"
@@ -101,6 +102,13 @@ func TestReplay(t *testing.T) {
 			path, _ := filepath.Abs(tc)
 			args := parseExec(t, path)
 			expected := expectedStderr(t, path)
+
+			if args.Process != "" {
+				_, err := osexec.LookPath(args.Process)
+				if err != nil {
+					t.Skipf("executable not found: %s %s", args.Process, err)
+				}
+			}
 
 			switch args.Command {
 			case "diff":
@@ -199,6 +207,7 @@ func expectedStderr(t *testing.T, dir string) string {
 
 type exec struct {
 	Command string `json:"command"`
+	Process string `json:"process"`
 }
 
 func parseExec(t *testing.T, dir string) exec {

--- a/internal/endtoend/endtoend_test.go
+++ b/internal/endtoend/endtoend_test.go
@@ -152,6 +152,9 @@ func cmpDirectory(t *testing.T, dir string, actual map[string]string) {
 		if filepath.Base(path) == "sqlc.json" {
 			return nil
 		}
+		if filepath.Base(path) == "exec.json" {
+			return nil
+		}
 		if strings.Contains(path, "/kotlin/build") {
 			return nil
 		}

--- a/internal/endtoend/testdata/process_plugin_sqlc_gen_json/exec.json
+++ b/internal/endtoend/testdata/process_plugin_sqlc_gen_json/exec.json
@@ -1,0 +1,3 @@
+{
+  "process": "sqlc-gen-json"
+}

